ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+        // coounterclockwise.
+        OGRGeometry *poGeom = OGRGeometry::FromHandle(hGeom);
+        if ((bExternalRing && !poGeom->toLinearRing()->isClockwise()) ||
+            (!bExternalRing && poGeom->toLinearRing()->isClockwise()))
+            hMMFeature.flag_VFG[hMMFeature.nIRing] |= MM_ROTATE_ARC;
+    }
+
+    hMMFeature.pNCoordRing[hMMFeature.nIRing] = OGR_G_GetPointCount(hGeom);
+
+    if (MMResizeMM_POINT2DPointer(&hMMFeature.pCoord, &hMMFeature.nMaxpCoord,
+                                  hMMFeature.nICoord +
+                                      hMMFeature.pNCoordRing[hMMFeature.nIRing],
+                                  MM_MEAN_NUMBER_OF_NCOORDS, 0))
+        return OGRERR_FAILURE;
+    if (MMResizeDoublePointer(&hMMFeature.pZCoord, &hMMFeature.nMaxpZCoord,
+                              hMMFeature.nICoord +
+                                  hMMFeature.pNCoordRing[hMMFeature.nIRing],
+                              MM_MEAN_NUMBER_OF_NCOORDS, 0))
+        return OGRERR_FAILURE;
+
+    for (int iPoint = 0;
+         (MM_N_VERTICES_TYPE)iPoint < hMMFeature.pNCoordRing[hMMFeature.nIRing];
+         iPoint++)
+    {
+        hMMFeature.pCoord[hMMFeature.nICoord].dfX = OGR_G_GetX(hGeom, iPoint);
+        hMMFeature.pCoord[hMMFeature.nICoord].dfY = OGR_G_GetY(hGeom, iPoint);
+        if (OGR_G_GetCoordinateDimension(hGeom) == 2)
+            hMMFeature.pZCoord[hMMFeature.nICoord] =
+                MM_NODATA_COORD_Z;  // Possible rare case
+        else
+        {
+            hMMFeature.pZCoord[hMMFeature.nICoord] = OGR_G_GetZ(hGeom, iPoint);
+            phMiraMonLayer->bIsReal3d = 1;
+        }
+
+        hMMFeature.nICoord++;
+    }
+    hMMFeature.nIRing++;
+    hMMFeature.nNRings++;
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                           MMLoadGeometry()                               */
+/*                                                                          */
+/*      Loads on a MiraMon object Feature all coordinates from feature      */
+/*                                                                          */
+/****************************************************************************/
+OGRErr OGRMiraMonLayer::MMLoadGeometry(OGRGeometryH hGeom)
+
+{
+    OGRErr eErr = OGRERR_NONE;
+    MM_BOOLEAN bExternalRing;
+
+    /* -------------------------------------------------------------------- */
+    /*      This is a geometry with sub-geometries.                         */
+    /* -------------------------------------------------------------------- */
+    int nGeom = OGR_G_GetGeometryCount(hGeom);
+
+    int eLT = LOG_ACTION(wkbFlatten(OGR_G_GetGeometryType(hGeom)));
+
+    if (eLT == wkbMultiPolygon || eLT == wkbPolyhedralSurface ||
+        eLT == wkbTIN || eLT == wkbTriangle)
+    {
+        for (int iGeom = 0; iGeom < nGeom; iGeom++)
+        {
+            OGRGeometryH poNewGeometry = OGR_G_GetGeometryRef(hGeom, iGeom);
```

Comment:
New implies that you've created something. Better use poSubGeometry. Also applies at line 1722
```suggestion
            OGRGeometryH poSubGeometry = OGR_G_GetGeometryRef(hGeom, iGeom);
```

Reply:
Done at other places also.